src/loader/bgw_launcher.c

```diff
@@ -481,21 +504,22 @@ message_restart_action(HTAB *db_htab, BgwMessage *message, VirtualTransactionId
 		terminate_background_worker(db_he->db_scheduler_handle);
 		wait_for_background_worker_shutdown(db_he->db_scheduler_handle);
 	}
-	else if (!bgw_total_workers_increment())	/* we still need to increment
-												 * if we haven't found an
-												 * entry */
+	else
 	{
-		report_bgw_limit_exceeded();
-		bgw_message_send_ack(message, ACK_FAILURE);
-		return;
+		if (!bgw_total_workers_increment()) /* we still need to increment if
```

Comment:
NON-NIT: In `if(!found)` we should set `db_scheduler_handle = NULL` and `incremented = false`,  like we do in `populate_database_htab`. Based on a quick skim of the allocation code hash-buckets are not guaranteed to be zeroed, and based on the `hash_search` docs we can't rely zeroing on anyway.

Reply:
Yep, same as above. fixed.